packages/testing/src/api/directive.ts

```diff
@@ -32,6 +32,50 @@ interface DirectiveMocker {
   (directive: TransformerDirective, executionMock: TransformerMock): any
 }
 
+/**
+ *
+ * @description
+ * Used for writing directive tests for asynchronous directives e.g.
+ * - Transformer directives can be passed resolvedValue
+ * - Validator directives should check for errors thrown in certain situtations
+ *
+ * @example
+ *  const mockExecution = mockAsyncRedwoodDirective(myTransformer, {
+ *    context: currentUser,
+ *    resolvedValue: 'Original Value',
+ *  })
+ *
+ *  expect(mockExecution).not.toThrow()
+ *  expect(mockExecution()).toEqual('Transformed Value')
```

Comment:
It looks like this example is still for the synchronous version? And is it `mockedResolvedValue` or `resolvedValue`?

```suggestion
 *  const mockExecution = mockAsyncRedwoodDirective(myTransformer, {
 *    context: currentUser,
 *    mockedResolvedValue: 'Original Value',
 *  })
 *
 *  await expect(mockExecution).resolves.not.toThrow()
 *  await expect(mockExecution()).resolves.toEqual('Transformed Value')
```

I'm not 100% sure about the `await expect(mockExecution).resolves.not.toThrow()` matcher. I know it works (I ran it) but it feels like there should be a more elegant way to do it since by definition, if it resolves, it didn't throw.